src/silo/test/string_search.test.cpp

```diff
@@ -0,0 +1,138 @@
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <nlohmann/json.hpp>
+
+#include "silo/test/query_fixture.test.h"
+
+using silo::ReferenceGenomes;
+using silo::config::DatabaseConfig;
+using silo::config::ValueType;
+using silo::test::QueryTestData;
+using silo::test::QueryTestScenario;
+
+namespace {
+
+const std::string TEST_COLUMN = "test_column";
+const std::string INDEXED_TEST_COLUMN = "indexed_test_column";
+
+nlohmann::json createDataEntry(std::string primary_key, std::string test_column_value) {
+   return {
+      {"metadata",
+       {{"primaryKey", primary_key},
+        {TEST_COLUMN, test_column_value},
+        {INDEXED_TEST_COLUMN, test_column_value}}},
+      {"alignedNucleotideSequences", {{"segment1", nullptr}}},
+      {"unalignedNucleotideSequences", {{"segment1", nullptr}}},
+      {"alignedAminoAcidSequences", {{"gene1", nullptr}}},
+      {"nucleotideInsertions", {{"segment1", {}}}},
+      {"aminoAcidInsertions", {{"gene1", {}}}}
+   };
+}
+
+const std::vector<nlohmann::json> DATA = {
```

Comment:
I think a test with null values would be helpful

Reply:
Yes, good point :+1:

Reply:
Done. I also added two more tests for queries that should fail: an invalid regex and null as search vallue.